docs/enhancements/install_logs.md

```diff
@@ -0,0 +1,92 @@
+# Store OpenShift Install Logs in S3 Compatible Storage
+
+
+## Summary
+
+The purpose of this enhancement is to change how Hive stores OpenShift install logs. Currently Hive stores OpenShift install logs in a persistent volume (PV).
+
+This enhancement changes the OpenShift install log location from placing them in PVs to placing them in S3 compatible object storage.
+
+
+## Motivation
+Storing OpenShift install logs inside of a PV works, but has several drawbacks:
+1. The cluster running Hive must have PVs enabled (with or without storage classes). This, for example, affects developer Hive instances.
+1. The number of OpenShift install pods is limited by the number of available PVs. Hive could exhaust a cluster's available PVs by doing too many OpenShift installs.
+1. Hive users must get the OpenShift install log through Hive instead of going directly to the storage. When OpenShift install logs are written to S3, then a Hive administrator can go directly to S3 to get the logs.
+1. OpenShift install logs are relatively small, but minimum PV sizes are usually large. For instance, in AWS, the minimum size of a PV is 1 GiB, but OpenShift install logs tend to be less than 100 MiB.
+
+### Goals
+Store OpenShift install logs inside of S3 object storage.
+
+### Non-Goals
+None
+
+## Open Questions
+1. What is the best way to pass the S3 parameters from the Hive operator to Hive controllers to the OpenShift install pod?
+   1. MUST allow for other object storage machanisms to be added in the future.
+   1. We'll look to how other projects like Velero do this as a reference.
+1. What object should store the URL of the OpenShift install log in S3?
+   1. Something like ClusterProvision.Status.S3.URI.
+
+## Proposal
+Change Hive to store OpenShift install logs in S3 compatible object storage instead of persistent volumes.
+
+Ensure that future enhancements can use similar mechanisms to upload to other object storage systems.
+
+## User Stories
+
+### Story 1
+As a Hive administrator,
+I want OpenShift install logs to be stored in S3 compatible object storage,
+So that simultaneous Hive triggered OpenShift installs can scale with available compute and memory instead of PVs.
+
+### Story 2
+As an Hive Developer,
+I want OpenShift install logs to be stored in S3 compatible object storage,
+So that while developing, I can use Minio to capture OpenShift install logs.
+
+
+## Implementation Details / Notes / Constraints
+1. Configure S3 URL and credentials in HiveConfig. Have these passed down into OpenShift install pods.
+  1. If no S3 information is given, do not gather OpenShift install logs. This will use the functionality already present for skipGatherLogs.
+1. From the OpenShift install pods, upload the OpenShift install log to S3.
+  1. Organize install logs using clustername, namespace. Then, name the install log with the cluster provision name. Use a pattern like: <clustername>-<namespace>/<clusterprovisionname>-install-log.tgz
```

Comment:
Does the clusterprovisionname include the install attempt #? Do we store install logs for failed installs?

Reply:
They do include the attempt number, ex: dgood-dev-0-vm2hj. We definitely will store the logs from failed attempts.